api/v1/canary_types.go

```diff
@@ -52,10 +52,12 @@ type CanarySpec struct {
 	Helm           []HelmCheck           `yaml:"helm,omitempty" json:"helm,omitempty"`
 	Namespace      []NamespaceCheck      `yaml:"namespace,omitempty" json:"namespace,omitempty"`
 	Redis          []RedisCheck          `yaml:"redis,omitempty" json:"redis,omitempty"`
-	IconURL        string                `yaml:"iconURL,omitempty" json:"iconURL,omitempty"`
-	Interval       uint64                `yaml:"interval,omitempty" json:"interval,omitempty"`
-	Severity       string                `yaml:"severity,omitempty" json:"severity,omitempty"`
-	Owner          string                `yaml:"owner,omitempty" json:"owner,omitempty"`
+	// Because the EC2 check attempts to clean up stale instances at the start of the check, only one EC2Check can be scoped per Canary instance
+	EC2      EC2Check `yaml:"ec2,omitempty" json:"ec2,omitempty"`
```

Comment:
It should only be cleaning up instances created by that check ?

Reply:
the instances are tagged with the name of the Canary, so if there are multiple checks per canary, there isn't a unique identifier to tell which check within the canary is which.